src/config.ts

```diff
@@ -41,32 +41,58 @@ export class MirrordConfigManager {
     private constructor() {
         this.fileListeners = [];
 
-        this.fileListeners.push(vscode.workspace.onDidDeleteFiles(event => {
+        this.fileListeners.push(vscode.workspace.onDidDeleteFiles(async event => {
             const activeConfigDeleted = event.files.find(file => file.path === this.active?.path) !== undefined;
-            if (!activeConfigDeleted) {
+            if (activeConfigDeleted) {
+                new NotificationBuilder()
+                    .withMessage("removed active mirrord configuration")
+                    .withDisableAction("promptActiveConfigRemoved")
+                    .warning();
+
+                this.setActiveConfig(undefined);
+
                 return;
             }
 
-            new NotificationBuilder()
-                .withMessage("removed active mirrord configuration")
-                .withDisableAction("promptActiveConfigRemoved")
-                .warning();
+            if (this.active) {
+                try {
+                    await vscode.workspace.fs.stat(this.active);
+                } catch (e) {
+                    new NotificationBuilder()
+                        .withMessage("removed folder containing active mirrord configuration")
+                        .withDisableAction("promptActiveConfigRemoved")
+                        .warning();
 
-            this.setActiveConfig(undefined);
+                    this.setActiveConfig(undefined);
+                }
+            }
         }));
 
-        this.fileListeners.push(vscode.workspace.onDidRenameFiles(event => {
+        this.fileListeners.push(vscode.workspace.onDidRenameFiles(async event => {
             const activeConfigMoved = event.files.find(file => file.oldUri.path === this.active?.path);
-            if (!activeConfigMoved) {
+            if (activeConfigMoved) {
+                new NotificationBuilder()
+                    .withMessage(`moved active mirrord configuration to ${vscode.workspace.asRelativePath(activeConfigMoved.newUri)}`)
+                    .withDisableAction("promptActiveConfigMoved")
+                    .warning();
+
+                this.setActiveConfig(activeConfigMoved.newUri);
+
                 return;
             }
 
-            new NotificationBuilder()
-                .withMessage("removed active mirrord configuration")
-                .withDisableAction("promptActiveConfigRemoved")
-                .warning();
+            if (this.active) {
+                try {
+                    await vscode.workspace.fs.stat(this.active);
+                } catch (e) {
+                    new NotificationBuilder()
+                        .withMessage("removed active mirrord configuration")
+                        .withDisableAction("promptActiveConfigRemoved")
+                        .warning();
 
-            this.setActiveConfig(undefined);
+                    this.setActiveConfig(undefined);
+                }
+            }
```

Comment:
What do you think about unifying the cases of file moved and ancestor moved?
We could `find` a file that is "a prefix" of the active path, and if it exists, the new active path would be the new renamed path plus the rest of the old active path after the prefix.
For example: `active.path` is  `/my/active/config`. We found a renamed file `/my/ -> /our/`, then the new path active path would be `/our/active/config`.
If we found a renamed file `/my/active/config -> /my/active/config.json` the new active path would just be the new path of the renamed file.
Then we would also have the good UX in the renamed dir case, with less LOC.

Reply:
Done! :)